.github/workflows/monthly-minor-release.yml

```diff
@@ -0,0 +1,44 @@
+name: Generate monthly minor release(s)
+on:
+  # Run first Monday of the month at 8AM PST / 9AM PDT
+  schedule:
+    - cron: '0 16 1-7 * 1'
+jobs:
+  generate-version-matrix:
+    name: Generate version matrix
+    runs-on: ubuntu-latest
+    if: github.repository == 'aws/sagemaker-distribution'
+    outputs:
+      matrix: ${{ steps.gen-mat.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Generate base patch version matrix
+        id: gen-mat
+        # Output looks like: matrix={"version":["0.11.0","1.4.1"]}
+        # For each major, get highest minor version. Then get highest patch
+        # for that minor. Use this patch as base version.
```

Comment:
On mondays where both the minor and patch workflows create new PRs, the patch PR would be created first but when the minor PR is created, the highest patch version it sees would still be from the week before right? If so, is that okay?

Reply:
Correct. The new patch version would still have the same minor version for all of the packages as previous patch version. So using either as base for new minor, would lead to the same upperbounds.